File: gateway/src/dtos/vault.dto.js
class VaultDTO {
    static validateCreate(body) {
        if (!body.dataType || typeof body.dataType !== 'string' || body.dataType.trim() === '') {
            throw new Error('dataType is required and must be a non-empty string');
        }
        if (!body.data || typeof body.data !== 'string' || body.data.trim() === '') {
            throw new Error('data is required and must be a non-empty string');
        }
        if (body.data.length > 50000) {
            throw new Error('data payload exceeds maximum allowed size');
        }
        return {
            dataType: body.dataType.trim(),
            data: body.data.trim()
        };
    }
}

module.exports = VaultDTO;

File: core/internal/dtos/vault_dto.go
package dtos

import (
	"errors"
	"strings"
)

type CreateVaultDataRequest struct {
	DataType string `json:"dataType"`
	Data     string `json:"data"`
}

func (r *CreateVaultDataRequest) Validate() error {
	r.DataType = strings.TrimSpace(r.DataType)
	r.Data = strings.TrimSpace(r.Data)
	if r.DataType == "" || r.Data == "" {
		return errors.New("dataType and data are required")
	}
	if len(r.Data) > 50000 {
		return errors.New("data payload exceeds maximum allowed size")
	}
	return nil
}

type VaultDataResponse struct {
	ID        string `json:"id"`
	DataType  string `json:"dataType"`
	Data      string `json:"data,omitempty"`
	CreatedAt string `json:"createdAt"`
}
